### PR TITLE
entrezdirect/esearch module with tests

### LINKED ISSUE #1771: new module: entrezdirect/esearch
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by alxndrdiaz**
## PR checklist

Closes #1771 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/entrezdirect/esearch/main.nf [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        'https://depot.galaxyproject.org/singularity/entrez-direct:13.9--pl526h375a9b1_1':
+        'quay.io/biocontainers/entrez-direct:13.9--pl526h375a9b1_1' }"
+
+    input:
+    tuple val(meta), val(database), val(term)
+    val sort_by
+    val date_type
+    val min_date
+    val max_date
+    val spell_check
```
[the remark this answers is not in the document]
Comment (priyanka-surana) [reply modules/entrezdirect/esearch/main.nf:16]: Do we really need these optional values? Why can't they be part of `ext.args`?

[the remark this answers is not in the document]
Comment (muffato) [reply modules/entrezdirect/esearch/main.nf:16]: Good point, and that would address https://github.com/nf-core/modules/pull/1807#discussion_r910839779

**Comment by alxndrdiaz** [comment]
For check: Pytest-workflow / entrezdirect/esearch conda (pull_request)  I have been unable to reproduce the error, pytest works fine locally.

File: modules/entrezdirect/esearch/main.nf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    """
+    esearch \\
+        ${args} \\
+        -db $database \\
+        -query $term \\
+        > ${prefix}.esearch.xml
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        esearch: \$(echo \$(esearch -version 2>&1) | sed 's/^esearch //; s/Using.*\$//' ))
```
Comment (muffato) [inline modules/entrezdirect/esearch/main.nf:32]:
```suggestion
        esearch: \$(esearch -version)
```

**Review by muffato — CHANGES_REQUESTED** [review]
Same as in #1833

**Review by jfy133 — APPROVED** [review]
LGTM!

**Review by muffato — COMMENTED** [review]
Looks good to me as well. Just make sure you rebase/merge as necessary to resolve the conflict in `pytest_modules.yml` (likely due to the merge of #1833).

Also, why can't you use MD5 checksums in `test.yml` ?

**Review by muffato — APPROVED** [review]

**Review by jfy133 — COMMENTED** [review]
I'm not sure what's changed but your output files don't contain some of your strings in all cases:

```
FAILED tests/modules/entrezdirect/esearch/test.yml::entrezdirect esearch test_entrezdirect_esearch_pubmed::output/entrezdirect/test_pubmed.xml::content::contains '<ENTREZ_DIRECT>'
FAILED tests/modules/entrezdirect/esearch/test.yml::entrezdirect esearch test_entrezdirect_esearch_pubmed::output/entrezdirect/test_pubmed.xml::content::contains '<Db>pubmed</Db>'
FAILED tests/modules/entrezdirect/esearch/test.yml::entrezdirect esearch test_entrezdirect_esearch_genome::output/entrezdirect/test_genome.xml::content::contains '<ENTREZ_DIRECT>'
FAILED tests/modules/entrezdirect/esearch/test.yml::entrezdirect esearch test_entrezdirect_esearch_genome::output/entrezdirect/test_genome.xml::content::contains '<Db>genome</Db>'
FAILED tests/modules/entrezdirect/esearch/test.yml::entrezdirect esearch test_entrezdirect_esearch_assembly::output/entrezdirect/test_assembly.xml::content::contains '<ENTREZ_DIRECT>'
FAILED tests/modules/entrezdirect/esearch/test.yml::entrezdirect esearch test_entrezdirect_esearch_assembly::output/entrezdirect/test_assembly.xml::content::contains '<Db>assembly</Db>'
```

**Review by alxndrdiaz — COMMENTED** [review]
Removed files from wrong merge.

**Comment by muffato** [comment]
Hi @alxndrdiaz . Same issue as in the other commit. The right branch to start from is `nf-core_master`, not `main`

**Pull request closed without merging**